recipes/gcc/all/conanfile.py

```diff
@@ -1,126 +1,193 @@
+
+    def build_requirements(self):
+        if self.info.settings.os == "Linux":
+            # binutils recipe is broken for Macos, and Windows uses tools
+            # distributed with msys/mingw
+            self.tool_requires("binutils/2.38")
+        self.tool_requires("flex/2.6.4")
 
     def requirements(self):
         self.requires("mpc/1.2.0")
         self.requires("mpfr/4.1.0")
-        self.requires("gmp/6.2.0")
-        self.requires("zlib/1.2.11")
+        self.requires("gmp/6.2.1")
+        self.requires("zlib/1.2.13")
+        self.requires("isl/0.24")
 
-    def configure(self):
-        if self.settings.os == "Windows":
-            raise ConanInvalidConfiguration("Windows builds aren't supported (yet), sorry")
-        if tools.cross_building(self.settings):
-            raise ConanInvalidConfiguration("no cross-building support (yet), sorry")
+    def package_id(self):
+        del self.info.settings.compiler
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = "gcc-%s" % self.version
-        os.rename(extracted_dir, self._source_subfolder)
+    def validate_build(self):
+        if is_msvc(self):
+            raise ConanInvalidConfiguration("GCC can't be built with MSVC")
+
+    def validate(self):
+        if self.info.settings.os == "Windows":
+            raise ConanInvalidConfiguration(
+                "Windows builds aren't currently supported. Contributions to support this are welcome."
+            )
+        if self.info.settings.os == "Macos":
+            # FIXME: This recipe should largely support Macos, however the following
+            # errors are present when building using the c3i CI:
+            # clang: error: unsupported option '-print-multi-os-directory'
+            # clang: error: no input files
+            raise ConanInvalidConfiguration(
+                "Macos builds aren't currently supported. Contributions to support this are welcome."
+            )
+        if cross_building(self):
+            raise ConanInvalidConfiguration(
+                "Cross builds are not current supported. Contributions to support this are welcome"
+            )
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def generate(self):
+        # Ensure binutils and flex are on the path.
+        # TODO: Remove when conan 2.0 is released as this will be default behaviour
+        buildenv = VirtualBuildEnv(self)
+        buildenv.generate()
+
+        tc = AutotoolsToolchain(self)
+        tc.configure_args.append("--enable-languages=c,c++,fortran")
+        tc.configure_args.append("--disable-nls")
+        tc.configure_args.append("--disable-multilib")
+        tc.configure_args.append("--disable-bootstrap")
+        tc.configure_args.append(f"--prefix={self.package_folder}")
+        tc.configure_args.append(
+            f"--libexecdir={os.path.join(self.package_folder, 'bin', 'libexec')}"
```

Comment:
I believe this line can be deleted as setting --libexecdir to self.package_folder is unnecessary in Conan 2.0, where the package_folder is, as I understand it, determined dynamically during packaging. The new paradigm is to leave --libexecdir at its default value and then leverage DESTDIR to relocate the package content during the installation process that takes place in the package method.